File: Java/src/main/java/org/codecop/lcdnumbers/NumeralSystem.java
package org.codecop.lcdnumbers;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

/**
 * Splits a number into its digits according to its numeral system, e.g. Decimal System.
 */
public class NumeralSystem {

    private final int base;

    public NumeralSystem() {
        this(10);
    }

    public NumeralSystem(int base) {
        if (base < 2) {
            throw new IllegalArgumentException("Smallest base is binary; base=" + base);
        }

        this.base = base;
    }

    public List<Integer> digitsOf(int value) {
        if (value < 0) {
            throw new IllegalArgumentException("negative number " + value);
        }

        List<Integer> digits = new ArrayList<>();

        int remainder = value;
        while (remainder > 0) {
            digits.add(remainder % base);
            remainder /= base;
        }
        Collections.reverse(digits);

        return digits;
    }
}

File: Csharp-MsDependencyInjection/LcdNumbers/NumeralSystem.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Org.Codecop.Lcdnumbers
{
    /// <summary>Splits a number into its digits according to its numeral system, e.g. Decimal System.</summary>
    public class NumeralSystem : INumeralSystem
    {
        private readonly int @base;

        public NumeralSystem(int @base)
        {
            if (@base < 2)
            {
                throw new ArgumentException("Smallest base is binary; base=" + @base);
            }

            this.@base = @base;
        }

        public virtual IList<int> DigitsOf(int value)
        {
            if (value < 0)
            {
                throw new ArgumentException("negative number " + value);
            }

            IList<int> digits = new List<int>();

            int remainder = value;
            while (remainder > 0)
            {
                digits.Add(remainder % @base);
                remainder /= @base;
            }

            return digits.Reverse().ToList();
        }
    }
}
